dhcpv4/nclient4/client.go

```diff
@@ -233,10 +234,16 @@ func withBufferCap(n int) ClientOpt {
 
 // WithRetry configures the number of retransmissions to attempt.
 //
+// 0 retries means infinite retries.
```

Comment:
I understand "zero retries" as "try just once, then stop" - same as above with "tries" vs. "retries". I'd rather keep using the concept of retries here

Reply:
which also means that 0 is a valid number of retries, and "loop forever" should be achieved differently (any negative number? Exactly -1 ? )

Reply:
went back to retries, but i think 0 is kind of a useless value, so <= 0 is infinite

Reply:
How I see the API as a user:
* if retries is 0, I expect one attempt
* if retries is 1, I expect two attempts
* and so on

The current code treats retries as attempts, so 0 is not meaningful, but I still think there is an inconsistency between naming and behaviour, which could confuse the users. Hope it makes sense, and open to discuss :)

Reply:
hm, why marked as resolved?